### PR TITLE
[TTAHUB-3197/3248] Add complete, suspend, resume functionality to Training reports

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Builds on the work in #2291 - it depends on the flags "ownerComplete" and "pocComplete" being set to work properly, so you may have to massage your local data to get tests to work.

- Add "complete event" to training report cards on the landing page. Should only be present if 1) both of the above fields are true in the event's data, 2) event has at least one session, and 3) all event's sessions are "complete." Only shows for owners.
- Also added a button to the "view event" screen with the same specs as above
- Also: Owners and admins can suspend an In progress or Not started event, and resume a suspended event via an action in the training report card screen

## How to test
Validate the criteria and behavior as described above. Ensure it matches the mockup.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3197

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/pages/TrainingReports/components/EventCard.js [hunk 1]
```diff
@@ -55,6 +58,23 @@ function EventCard({
   || data.status === TRAINING_REPORT_STATUSES.SUSPENDED);
   const menuItems = [];
 
+  const canCompleteEvent = (() => {
+    if (!isOwner && !isNotCompleteOrSuspended) {
+      return false;
+    }
+
+    if (!trainingReportComplete) {
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/TrainingReports/components/EventCard.js:66]: I don't think we should have the ! not on this IF. We only want to add the 'Complete' option if it't not currently completed.

File: frontend/src/pages/TrainingReports/components/EventCard.js [hunk 2]
```diff
@@ -55,6 +58,23 @@ function EventCard({
   || data.status === TRAINING_REPORT_STATUSES.SUSPENDED);
   const menuItems = [];
 
+  const canCompleteEvent = (() => {
+    if (!isOwner && !isNotCompleteOrSuspended) {
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/TrainingReports/components/EventCard.js:62]: The ! notCompletedOrSuspended is also sort of confusing here (double not).

File: frontend/src/pages/TrainingReports/components/EventCard.js (hunk 1 above)
Comment (thewatermethod) [reply frontend/src/pages/TrainingReports/components/EventCard.js:66]: It's not the status, it's whether the event has been submitted. I can change the name if you find it confusing.

File: frontend/src/pages/TrainingReports/components/EventCard.js (hunk 2 above)
Comment (thewatermethod) [reply frontend/src/pages/TrainingReports/components/EventCard.js:62]: See above

File: frontend/src/pages/TrainingReports/components/EventCard.js (hunk 1 above)
Comment (AdamAdHocTeam) [reply frontend/src/pages/TrainingReports/components/EventCard.js:66]: Should I be able to complete a 'Completed' event?

image

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good pointed out some things but if they are correct we should be good to merge to the other branch.

**Comment by thewatermethod** [comment]
@AdamAdHocTeam I believe I addressed the issue - will merge, let me know if I need to revisit something within your branch.

**Pull request merged**